CMakeLists.txt

```diff
@@ -239,7 +242,7 @@ if ( ENABLE_TESTS )
   enable_testing()
 
   # emulate GNU Autotools `make check`
-  add_custom_target(check COMMAND ${CMAKE_CTEST_COMMAND})
+  add_custom_target(check COMMAND ${CMAKE_CTEST_COMMAND} -C $<CONFIG>)
```

Comment:
MSVS *always* sets a configuration, and outputs files in configuration specific directories, so you always need to pass the `-C` flag with the generator expression `$<CONFIG>` to get this target to work from within MSVS.